packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,692 @@
+      ..selectedValue = selectedValue
+      ..pressedValue = pressedValue;
+  }
+}
+
+class _RenderSegmentedControlElement extends RenderObjectElement {
+  _RenderSegmentedControlElement(_SegmentedControlRenderWidget segmentedControl)
+      : super(segmentedControl);
+
+  final List<Element> _slots = <Element>[];
+
+  @override
+  _SegmentedControlRenderWidget get widget => super.widget;
+
+  @override
+  _RenderSegmentedControl get renderObject => super.renderObject;
+
+  @override
+  void update(_SegmentedControlRenderWidget newWidget) {
+    super.update(newWidget);
+    assert(widget == newWidget);
+    for (int i = 0; i < widget.children.length; i += 1) {
+      _updateChild(widget.children.elementAt(i), i);
+    }
+  }
+
+  void _updateChild(Widget widget, int slot) {
+    final Element oldChild = _slots.elementAt(slot);
+    final Element newChild = updateChild(oldChild, widget, slot);
+    if (oldChild != null) {
+      _slots.remove(oldChild);
+    }
+    if (newChild != null) {
+      _slots.insert(slot, newChild);
+    }
+  }
+
+  @override
+  void visitChildren(ElementVisitor visitor) {
+    _slots.forEach(visitor);
+  }
+
+  @override
+  void forgetChild(Element child) {
+    assert(_slots.contains(child));
+    _slots.remove(child);
+  }
+
+  @override
+  void insertChildRenderObject(RenderObject child, dynamic value) {
+    assert(child is RenderBox);
+    assert(value is int);
+    assert(value <= _slots.length);
+    renderObject._setBox(value, child);
+    assert(renderObject._slots.contains(child));
+  }
+
+  @override
+  void removeChildRenderObject(RenderObject child) {
+    assert(child is RenderBox);
+    assert(renderObject._slots.contains(child));
+    renderObject._setBox(renderObject.boxes.indexOf(child), null);
+    assert(!renderObject._slots.contains(child));
+  }
+
+  @override
+  void moveChildRenderObject(RenderObject child, dynamic value) {
+    assert(false, 'not reachable');
+  }
+
+  @override
+  void mount(Element parent, dynamic newSlot) {
+    super.mount(parent, newSlot);
+    for (int i = 0; i < widget.children.length; i += 1) {
+      _mountChild(widget.children.elementAt(i), i);
+    }
+  }
+
+  void _mountChild(Widget widget, int slot) {
+    Element oldChild;
+    if (slot >= _slots.length) {
+      oldChild = null;
+    } else {
+      oldChild = _slots.elementAt(slot);
+    }
+    final Element newChild = updateChild(oldChild, widget, slot);
+    if (oldChild != null) {
+      _slots.remove(oldChild);
+    }
+    if (newChild != null) {
+      _slots.insert(slot, newChild);
+    }
+  }
+}
+
+class _RenderSegmentedControl extends RenderBox {
+  _RenderSegmentedControl({
+    @required this.children,
+    @required this.selectedValue,
+    @required this.pressedValue,
+    @required TextDirection textDirection,
+  }) : _textDirection = textDirection;
+
+  List<Widget> children;
+
+  ValueKey<int> selectedValue;
+  ValueKey<int> pressedValue;
+
+  final List<Key>  _hitTestLog = <Key>[];
```

Comment:
Since these variables don't have documentation, you can collapse them into one block (no need for the extra lines between each one).